### ISSUE TITLE
Yeoman Generator: options.componentType is ignored if options.environment is undefined

### ISSUE THREAD

**Issue by AJIXuMuK** [SharePoint/sp-dev-docs#2104]
#### Category
- [ ] Question
- [ ] Typo
- [x] Bug
- [ ] Additional article idea

#### Expected or Desired Behavior

In custom generator we can provide specific `componentType` without providing `environment`.

Example: we want to create custom generator for web parts. In that case we want to provide `componentType` but not `environment`. Like that:

```javascript
this.composeWith(require.resolve('@microsoft/generator-sharepoint/lib/generators/app'), {
componentType: 'webpart'
});
```

The issue exists because of app generator code:
```javascript
if (this.options.environment !== undefined) {
            this.config.set('componentType', this.options.environment === 'onprem' ? 'webpart' : this.options.componentType);
        }
```
#### Observed Behavior

`componentType` option is not used if `environment` option is not set.
#### Steps to Reproduce

- create custom Yeoman generator
- `this.composeWith(require.resolve('@microsoft/generator-sharepoint/lib/generators/app'), { componentType: 'webpart' })`

**Comment by StfBauer** [comment]
I double checked your issue and it seems that you pass in the component-type wrong. Since you pass in options in the form of command line arguments you may not use the SPFx internal variable names and options.
I verified this behaviour with the following options passed in to SPFx
```
{ 'skip-install': true,
  'component-type': 'webpart',
  'component-description': 'HelloWorld',
  'component-name': 'helloworld',
  'solution-name': 'HelloWorld' }
```

**Comment by AJIXuMuK** [comment]
@StfBauer it doesn't work as well.
```javascript
const options = JSON.parse(JSON.stringify(this.options));
        options['component-type'] = 'webpart';
        options.framework = 'none';

        this.composeWith(require.resolve(`@microsoft/generator-sharepoint/lib/generators/app`), options);
```

**Comment by mcmynn83** [comment]
This is fixed in 1.5.1.

**Comment by VesaJuvonen** [comment]
Let's keep this open until we get a confirmation that 1.5.1 will actually fix this.

**Comment by AJIXuMuK** [comment]
Confirming that it is fixed in 1.5.1.

Thanks for the effort!
